nix/modules/devShells.nix

```diff
@@ -25,14 +25,21 @@
       devShells = {
         default = self'.devShells.holonix;
         holonix = pkgs.lib.makeOverridable
-          ({ holochainOverrides }: pkgs.mkShell {
+          ({ holochainOverrides }: (if inputs.versions.stub or false then pkgs.mkShell {
+            shellHook = ''
+                echo "This Holochain version is not supported on the current Holonix version. Please migrate to the new Holonix"
+                echo "See the instructions here: https://developer.holochain.org/resources/upgrade/upgrade-new-holonix/"
+                echo "Alternatively, you can use the quickstart guide provided with the updated Holonix: https://github.com/holochain/holonix"
+                exit 1;
```

Comment:
The dummy shell takes no inputs so it won't build anything and just prints a message before exiting.